src/signTx.c

```diff
@@ -1294,10 +1151,88 @@ static void signTx_handleMetadataAPDU(uint8_t p2, uint8_t* wireDataBuffer, size_
 	signTx_handleMetadata_ui_runStep();
 }
 
+// ============================== VALIDITY INTERVAL START ==============================
+
+enum {
+	HANDLE_VALIDITY_INTERVAL_START_STEP_DISPLAY = 900,
+	HANDLE_VALIDITY_INTERVAL_START_STEP_RESPOND,
+	HANDLE_VALIDITY_INTERVAL_START_STEP_INVALID,
+};
+
+static void signTx_handleValidityInterval_ui_runStep()
```

Comment:
UX-wise I find it a bit counterintuitive that validity interval start is being confirmed after TTL, but I guess reversing the order is not really an option given that it's coupled with tx serialization :/

Also, it may make sense to explicitly notify the user that the tx has no ttl/validity start

I'm wondering if it would not make sense to (eventually) move ttl/validity start directly to the init APDU to make the UX around those values more intuitive and remove that "machinery" just for two numbers

Reply:
I don't understand why they did not just replace ttl with validity interval (in the same way as coin in output is replaced by coin / [coin, multiasset]). That would fix all ui problems and also look more natural.

On one hand, we don't have much time to do changes; on the other, why show the users the "wrong" user flow with validity interval if we can avoid it?...